addons/microsoft_calendar/models/calendar.py

```diff
@@ -217,6 +222,18 @@ def _get_organizer_user_change_info(self, values):
                     partner_ids.append(command[2].get('partner_id'))
         return sender_user, partner_ids
 
+    def _update_attendee_status(self, attendee_ids):
+        # Merge current status from attendees_ids with new attendees values for avoiding their info loss in write().
+        # Create a dict getting the state of each attendee received from attendee_ids variable and then update their state.
+        state_by_attendee = {}
```

Comment:
You're not grouping by attendee, but by partner
```suggestion
        state_by_partner = {}
```